pkg/plugins/policies/meshhttproute/api/v1alpha1/meshhttproute.go

```diff
@@ -103,10 +104,46 @@ type HeaderModifier struct {
 	Remove []string `json:"remove,omitempty"`
 }
 
+// PreciseHostname is the fully qualified domain name of a network host. This
+// matches the RFC 1123 definition of a hostname with 1 notable exception that
+// numeric IP addresses are not allowed.
+//
+// Note that as per RFC1035 and RFC1123, a *label* must consist of lower case
+// alphanumeric characters or '-', and must start and end with an alphanumeric
+// character. No other punctuation is allowed.
+//
+// +kubebuilder:validation:MinLength=1
+// +kubebuilder:validation:MaxLength=253
+// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
+type PreciseHostname string
+
+// PortNumber defines a network port.
+//
+// +kubebuilder:validation:Minimum=1
+// +kubebuilder:validation:Maximum=65535
+type PortNumber int32
+
+type RequestRedirect struct {
+	Scheme   *string          `json:"scheme,omitempty"`
+	Hostname *PreciseHostname `json:"hostname,omitempty"`
+	// Port is the port to be used in the value of the `Location`
+	// header in the response.
+	// When empty, port (if specified) of the request is used.
+	//
+	Port *PortNumber `json:"port,omitempty"`
+	// StatusCode is the HTTP status code to be used in response.
+	//
+	// +kubebuilder:default=302
+	// +kubebuilder:validation:Enum=301;302;303;307;308
+	StatusCode *int `json:"statusCode,omitempty"`
+}
+
 type Filter struct {
-	Type                   FilterType      `json:"type"`
-	RequestHeaderModifier  *HeaderModifier `json:"requestHeaderModifier,omitempty"`
-	ResponseHeaderModifier *HeaderModifier `json:"responseHeaderModifier,omitempty"`
+	Type                   FilterType       `json:"type"`
+	RequestHeaderModifier  *HeaderModifier  `json:"requestHeaderModifier,omitempty"`
+	ResponseHeaderModifier *HeaderModifier  `json:"responseHeaderModifier,omitempty"`
+	RequestRedirect        *RequestRedirect `json:"requestRedirect,omitempty"`
+	// TODO: add path to redirect after adding URL rewrite
```

Comment:
is it something that you forgot to do in this PR or is it something to be done in next PR?

Reply:
next PR

Reply:
I have it in the tracking issue